Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -560,7 +561,13 @@ private class MutableDictionary: BSONValue {
         }
         set(newValue) {
             keys.append(key)
-            values.append(newValue)
+            if let newValue = newValue {
+                values.append(newValue)
+            } else {
+                guard let index = keys.index(of: key) else { return }
```

Comment:
Since we're always doing `keys.append(key)`, can this lead to a situation where our string exists twice in the `keys` array and we only remove the first occurrence that this `index(of:)` call finds?

Reply:
That is actually possible, I think. But since I've already moved the `append` call inside the non-`nil` case, this should no longer be an issue.